tests/qos/test_buffer.py

```diff
@@ -1227,59 +1263,151 @@ def test_exceeding_headroom(duthosts, rand_one_dut_hostname, conn_graph_facts, p
+                                                'Fetch the longest possible cable length',
+                                                ['Update speed .* and cable length .* for port .* failed, accumulative headroom size exceeds the limit',
+                                                 'Unable to update profile for port .*. Accumulative headroom size exceeds limit',
+                                                 'Failed to process table update',
+                                                 'oid is set to null object id on SAI_OBJECT_TYPE_BUFFER_PROFILE',
+                                                 'Failed to remove buffer profile .* with type BUFFER_PROFILE_TABLE',
+                                                 'doTask: Failed to process buffer task, drop it'])
+        logging.info('[Find out the longest cable length the port can support]')
+        cable_length = 300
+        while True:
+            duthost.shell('config interface cable-length {} {}m'.format(port_to_test, cable_length))
+            expected_profile = 'pg_lossless_{}_{}m_profile'.format(original_speed, cable_length)
+            profile_applied = check_pg_profile(duthost, 'BUFFER_PG_TABLE:{}:3-4'.format(port_to_test), expected_profile, False)
+            if not profile_applied:
+                break
+            logging.debug('Cable length {} has been applied successfully'.format(cable_length))
+            cable_length += 100
+
+        # We've got the maximum cable length that can be applied on the port
+        violating_cable_length = cable_length
+        maximum_cable_length = cable_length - 100
+        logging.info('Got maximum cable length {}'.format(maximum_cable_length))
+
+        # Check whether there is the expected error message in the log
+        logging.info('Check whether the expected error message is found')
+        check_log_analyzer(loganalyzer, marker)
+
+        loganalyzer, marker = init_log_analyzer(duthost,
+                                                'Add addtional PGs',
+                                                ['Update speed .* and cable length .* for port .* failed, accumulative headroom size exceeds the limit',
+                                                 'Unable to update profile for port .*. Accumulative headroom size exceeds limit'])
+
+        maximum_profile_name = 'pg_lossless_{}_{}m_profile'.format(original_speed, maximum_cable_length)
+        maximum_profile = _compose_dict_from_cli(duthost.shell('redis-cli hgetall BUFFER_PROFILE_TABLE:{}'.format(maximum_profile_name))['stdout'].split())
+
+        # Config the cable length to the longest acceptable value and check the profile
+        logging.info('[Config the cable length to the longest acceptable value on the port]')
+        duthost.shell('config interface cable-length {} {}m'.format(port_to_test, maximum_cable_length))
+        check_pg_profile(duthost, 'BUFFER_PG_TABLE:{}:3-4'.format(port_to_test), maximum_profile_name)
```

Comment:
do you want to capture the return value of this func?

Reply:
We don’t need it because by default it will fail the test in case of a failure.
The only case it returns False when an error occurs is the beginning of the test where the maximum headroom is detected.